### ISSUE TITLE
Nothing shows in Homebridge App

### ISSUE THREAD

**Issue by brent7320** [birkir/homebridge-plugin-eufy-security#4]
I have installed the plugin in homebridge and supplied valid credentials, no accessories ever appear. Are there any additional instructions we should be following? I have restarted Homebridge multiple times, nothing ever appears. This is a Battery 2K Doorbell.

Plugin version - 0.0.3
Homebridge - v1.2.3
NPM - 6.14.7

12/1/2020, 8:17:03 PM] [EufySecurityHomebridgePlugin] credentials found. reusing them...

**Comment by birkir** [comment]
Try the 0.1.1 version I just published. Start with `homebridge -D` and see if there are any logs

**Comment by Techlover22** [comment]
> I have installed the plugin in homebridge and supplied valid credentials, no accessories ever appear. Are there any additional instructions we should be following? I have restarted Homebridge multiple times, nothing ever appears. This is a Battery 2K Doorbell.
> 
> Plugin version - 0.0.3
> Homebridge - v1.2.3
> NPM - 6.14.7
> 
> 12/1/2020, 8:17:03 PM] [EufySecurityHomebridgePlugin] credentials found. reusing them...

I'm having the same issue too. I'm sending you a zip of the home bridge.log file
[homebridge.log.zip](https://github.com/birkir/homebridge-plugin-eufy-security/files/5626866/homebridge.log.zip)

**Comment by DaveM8686** [comment]
I've updated to 0.1.1 and I'm still having this exact issue, including the part where the only thing the log says is [EufySecurityHomebridgePlugin] credentials found. reusing them...

**Comment by ktsi** [comment]
Hi!

I have used a previous version of the plug-in with the exact same Dort bell. It worked after installing twice.

**Comment by DaveM8686** [comment]
I've uninstalled and reinstalled so many times. I've installed while it's already installed. I've tried from the terminal and from the HomeBridge UI. I've restarted the server repeatedly. Set up a new guest account for the doorbell, turned off 2FA, nothing. 

I'm running HomeBridge on a Synology NAS through Docker, if that makes a difference.

**Comment by brent7320** [comment]
> > I have installed the plugin in homebridge and supplied valid credentials, no accessories ever appear. Are there any additional instructions we should be following? I have restarted Homebridge multiple times, nothing ever appears. This is a Battery 2K Doorbell.
> 
> > 
> 
> > Plugin version - 0.0.3
> 
> > Homebridge - v1.2.3
> 
> > NPM - 6.14.7
> 
> > 
> 
> > 12/1/2020, 8:17:03 PM] [EufySecurityHomebridgePlugin] credentials found. reusing them...
> 
> 
> 
> I'm having the same issue too. I'm sending you a zip of the home bridge.log file
> 
> [homebridge.log.zip](https://github.com/birkir/homebridge-plugin-eufy-security/files/5626866/homebridge.log.zip)
> 
> 

You might want to take this attachment down and change your eufy password ASAP as it was in the above log file.

**Comment by brent7320** [comment]
I ran homebridge in debug mode and below is the only additional information displayed in the logs. 

[12/2/2020, 4:35:38 AM] [EufySecurityHomebridgePlugin] Initializing EufySecurityHomebridgePlugin platform... [12/2/2020, 4:35:38 AM] [EufySecurityHomebridgePlugin] Config { username: 'xxx@xxx.com', password: 'xxxxx', platform: 'EufySecurityHomebridgePlugin' } [12/2/2020, 4:35:38 AM] [EufySecurityHomebridgePlugin] Finished initializing platform: undefined

**Comment by dshafik** [comment]
I'm having this exact same issue. I spent some time debugging with log files and this like seems to be where it bails:

https://github.com/birkir/homebridge-plugin-eufy-security/blob/master/src/platform.ts#L82

I cannot get passed that. I do currently have 2FA turned off for testing.

**Comment by brent7320** [comment]
> I'm having this exact same issue. I spent some time debugging with log files and this like seems to be where it bails:
> 
> 
> 
> https://github.com/birkir/homebridge-plugin-eufy-security/blob/master/src/platform.ts#L82
> 
> 
> 
> I cannot get passed that. I do currently have 2FA turned off for testing. I have a wired doorbell, but I don't think it matters in this case.

I initially had 2FA turned on as well, but disabled for testing. Not sure if that has something to do with it not working or not.

**Comment by DaveM8686** [comment]
Same! I had 2FA on at first, and it won't work for me. Even rolling back my config file and completely setting up the plug-in from scratch, it still says it found existing credentials and is reusing them. I think this is the problem.

**Comment by raphb83** [comment]
I got same issue. Installed version 0.1.1 installing twice did not held.

Startup logs says:

2.12.2020, 15:47:47 (node:9291) UnhandledPromiseRejectionWarning: Error: Request failed: https://security-app-eu.eufylife.com/v1/passport/login -> 26052 - need validate code
2.12.2020, 15:47:47 at /home/hoobs/.hoobs/node_modules/eufy-node-client/build/http/http.utils.js:24:15
2.12.2020, 15:47:47 at Generator.next (<anonymous>)
2.12.2020, 15:47:47 at fulfilled (/home/hoobs/.hoobs/node_modules/eufy-node-client/build/http/http.utils.js:5:58)
2.12.2020, 15:47:47 at runMicrotasks (<anonymous>)
2.12.2020, 15:47:47 at processTicksAndRejections (internal/process/task_queues.js:97:5)
2.12.2020, 15:47:47 (node:9291) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
2.12.2020, 15:47:47 (node:9291) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by DaveM8686** [comment]
I actually tried deleting the "eufy-security-credentials.json" file and running again.  It found the new credentials in this case, but still nothing showing in Home.

**Comment by dshafik** [comment]
@raphb83 do you have 2 Factor Auth (2FA) enabled? I believe I saw the same behavior when I did. Currently 2FA isn't supported.

**Comment by dshafik** [comment]
@DaveM8686 same here. I had hoped it was bad credentials due to the 2FA issue, but it didn't help.

**Comment by Techlover22** [comment]
> @raphb83 do you have 2 Factor Auth (2FA) enabled? I believe I saw the same behavior when I did. Currently 2FA isn't supported.

I'm having the same issue but without 2FA...

**Comment by dshafik** [comment]
With some more debugging and patience (takes 5+ minutes) I finally got an error out of: https://github.com/birkir/homebridge-plugin-eufy-security/blob/master/src/platform.ts#L82

```json
{ RequestError: read ECONNRESET
    at ClientRequest.request.once (/homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/core/index.js:953:111)
    at Object.onceWrapper (events.js:286:20)
    at ClientRequest.emit (events.js:203:15)
    at ClientRequest.origin.emit.args (/homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/@szmarczak/http-timer/dist/source/index.js:39:20)
    at TLSSocket.socketErrorListener (_http_client.js:401:9)
    at TLSSocket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at TLSWrap.onStreamRead (internal/stream_base_commons.js:111:27)
  name: 'RequestError',
  code: 'ECONNRESET',
  timings:
   { start: 1606905673656,
     socket: 1606905673656,
     lookup: 1606905673674,
     connect: 1606905673762,
     secureConnect: 1606905675424,
     upload: 1606905673657,
     response: undefined,
     end: undefined,
     error: 1606905675794,
     abort: undefined,
     phases:
      { wait: 0,
        dns: 18,
        tcp: 88,
        tls: 1662,
        request: NaN,
        firstByte: undefined,
        download: undefined,
        total: 2138 } } }
```

This is a pretty opaque error in that it just means that the Eufy server abruptly closed the connection (probably due to a timeout after 5m!).

Because I catch that exception and return from the function, it then tries to find the devices, and results in this:

```
UnhandledPromiseRejectionWarning: RequestError: getaddrinfo EAI_AGAIN security-app.eufylife.com security-app.eufylife.com:443
    at ClientRequest.request.once (/homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/core/index.js:953:111)
    at Object.onceWrapper (events.js:286:20)
    at ClientRequest.emit (events.js:203:15)
    at ClientRequest.origin.emit.args (/homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/@szmarczak/http-timer/dist/source/index.js:39:20)
    at TLSSocket.socketErrorListener (_http_client.js:401:9)
    at TLSSocket.emit (events.js:198:13)
    at emitErrorNT (internal/streams/destroy.js:91:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:59:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:56:26)
(node:6518) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:6518) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This is a DNS issue. So I figured out the IP for security-app.eufylife.com (it's a CNAME to AWS, which then resolves to multiple IPs, I picked the first one), and added it to my `/etc/hosts` and now I get:

```json
{ HTTPError: Response code 401 (Unauthorized)
    at Request.request.once (/homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/as-promise/index.js:117:42)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  name: 'HTTPError',
  code: undefined,
  timings:
   { start: 1606957517771,
     socket: 1606957517771,
     lookup: 1606957517771,
     connect: 1606957517880,
     secureConnect: 1606957517984,
     upload: 1606957517771,
     response: 1606957518074,
     end: 1606957518074,
     error: undefined,
     abort: undefined,
     phases:
      { wait: 0,
        dns: 0,
        tcp: 109,
        tls: 104,
        request: NaN,
        firstByte: 303,
        download: 0,
        total: 303 } } }`

and when trying to find devices:

```
UnhandledPromiseRejectionWarning: HTTPError: Response code 401 (Unauthorized)
    at Request.request.once (/homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/as-promise/index.js:117:42)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:7344) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:7344) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This might be caused by my docker-in-virtualbox-bridged-bridged networking, so I'd love to know if anyone else can confirm.

If you want to try yourself, edit `node_modules/homebridge-plugin-security-eufy/dist/platform.js` in your Homebridge installation directory, and look for this line around line 65 (I've added multiple debug lines above, not sure how many):

```javascript
await this.httpService.registerPushToken(fcmToken);
```

and replace with:

```javascript
        try {
                await this.httpService.registerPushToken(fcmToken);
        } catch (e) {
                this.log.debug("EXCEPTION!");
                this.log.debug(e);
                return;
        }
```

**Comment by brent7320** [comment]
I have added that debug statement, however I get no additional logs in my console after restarting homebridge. The last log relevant to Eufy is this. 

[12/3/2020, 2:16:59 AM] [EufySecurityHomebridgePlugin] Initializing EufySecurityHomebridgePlugin platform...
[12/3/2020, 2:16:59 AM] [EufySecurityHomebridgePlugin] Config { username: 'xxxxxxx',
  password: 'xxxxxx',
  platform: 'EufySecurityHomebridgePlugin' }
[12/3/2020, 2:16:59 AM] [EufySecurityHomebridgePlugin] Finished initializing platform: undefined

**Comment by brent7320** [comment]
> If you want to try yourself, edit `node_modules/homebridge-plugin-security-eufy/dist/platform.js` in your Homebridge installation directory, and look for this line around line 65 (I've added multiple debug lines above, not sure how many):
> 
> 
> 
> ```javascript
> 
> await this.httpService.registerPushToken(fcmToken);
> 
> ```
> 
> 
> 
> and replace with:
> 
> 
> 
> ```javascript
> 
>         try {
> 
>                 await this.httpService.registerPushToken(fcmToken);
> 
>         } catch (e) {
> 
>                 this.log.debug("EXCEPTION!");
> 
>                 this.log.debug(e);
> 
>                 return;
> 
>         }
> 
> ```

I was able to find my cached credential .Json file and I deleted it. After doing that and putting in the log statements you stated above, i was able to duplicate what you are seeing. I am not running this in docker though, this is homebridge running on an Ubuntu VM. 

```

[12/3/2020, 2:39:03 AM] [EufySecurityHomebridgePlugin] EXCEPTION!
[12/3/2020, 2:39:03 AM] [EufySecurityHomebridgePlugin] { HTTPError: Response code 401 (Unauthorized)
    at Request.request.once (/usr/lib/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/as-promise/index.js:117:42)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  name: 'HTTPError',
  code: undefined,
  timings:
   { start: 1606984743636,
     socket: 1606984743636,
     lookup: 1606984743664,
     connect: 1606984743703,
     secureConnect: 1606984743748,
     upload: 1606984743636,
     response: 1606984743794,
     end: 1606984743794,
     error: undefined,
     abort: undefined,
     phases:
      { wait: 0,
        dns: 28,
        tcp: 39,
        tls: 45,
        request: NaN,
        firstByte: 158,
        download: 0,
        total: 158 } } }
[12/3/2020, 2:39:03 AM] [EufySecurityHomebridgePlugin] Executed didFinishLaunching callback
(node:670387) UnhandledPromiseRejectionWarning: HTTPError: Response code 401 (Unauthorized)
    at Request.request.once (/usr/lib/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/as-promise/index.js:117:42)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:670387) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:670387) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

**Comment by raphb83** [comment]
> @raphb83 do you have 2 Factor Auth (2FA) enabled? I believe I saw the same behavior when I did. Currently 2FA isn't supported.

As far as i know i do not have 2FA enabled. Sorry about the question: where do i have to look if it is enabled?😬

**Comment by hicksbjacob** [comment]
I am having the same issue. I dont have 2FA enabled. after putting in my credentials and plug in nothing appears in accessories

**Comment by asgertler** [comment]
Also having this issue. On 0.1.1, just got my battery doorbell cam, no 2FA, but nothing shows up in Home.

**Comment by awhelme** [comment]
I have the same problem and super excited to use it! Thanks for building this!

**Comment by cmfrazier** [comment]
Getting this error when attempting to use the plugin 
![image](https://user-images.githubusercontent.com/20451193/101122804-f1af8a80-35b8-11eb-97e0-2b46005b3587.jpeg)

**Comment by cmfrazier** [comment]
I deleted my `eufy-security-credentials.json` file as I had changed my password after turning off 2FA and I'm now getting this:
```
[12/4/2020, 10:37:59] [EufySecurityHomebridgePlugin] wait a short time (5sec)...
(node:30864) UnhandledPromiseRejectionWarning: HTTPError: Response code 401 (Unauthorized)
    at Request.<anonymous> (/usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/as-promise/index.js:117:42)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:30864) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
(node:30864) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by a-metasyntactic** [comment]
If it's a 2FA related problem, you can try creating an new account and inviting to have read only access to the cameras/doorbells you want to expose. Then use that new account's credentials for this plugin and you can keep 2FA enabled for your admin account.

I'd suggest that's a better and more secure way of setting this up anyway.

**Comment by DaveM8686** [comment]
> If it's a 2FA related problem, you can try creating an new account and inviting to have read only access to the cameras/doorbells you want to expose. Then use that new account's credentials for this plugin and you can keep 2FA enabled for your admin account.
> 
> I'd suggest that's a better and more secure way of setting this up anyway.

I’ve done this and it didn’t fix anything. I created a whole new account that never had 2FA, deleted the credentials file, uninstalled the plugin, then redid it all and it’s the same problem.

**Comment by birkir** [comment]
I added a option to disable push if you like (and fixed the issue that it blocks further code to run).

I also added some debug parameters so you should be able to report better logs for me.

**Comment by cmfrazier** [comment]
@birkir After updating and enabling debug
```
[12/5/2020, 12:23:32] [EufySecurityHomebridgePlugin] Initializing EufySecurityHomebridgePlugin platform...
[12/5/2020, 12:23:32] [EufySecurityHomebridgePlugin] Config {
  username: 'email-redacted',
  password: 'password-redacted',
  platform: 'EufySecurityHomebridgePlugin'
}
[12/5/2020, 12:23:32] [EufySecurityHomebridgePlugin] Finished initializing platform: undefined
[12/5/2020, 12:23:33] [EufySecurityHomebridgePlugin] push client disabled
[12/5/2020, 12:23:33] [EufySecurityHomebridgePlugin] Executed didFinishLaunching callback
(node:17243) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 shutdown listeners added to [HomebridgeAPI]. Use emitter.setMaxListeners() to increase limit
```

**Comment by raphb83** [comment]
> I added a option to disable push if you like (and fixed the issue that it blocks further code to run).
> 
> 
> 
> I also added some debug parameters so you should be able to report better logs for me.

Version 0.1.2 working with 2FA disabled and wired door bell😃
Great

**Comment by dshafik** [comment]
> I added a option to disable push if you like (and fixed the issue that it blocks further code to run).

What does this disable? Notifications?

> I also added some debug parameters so you should be able to report better logs for me.

Great!

Here's my current (redacted) logs:

```
[12/5/2020, 12:32:44 PM] [EufySecurityHomebridgePlugin] Initializing EufySecurityHomebridgePlugin platform...
[12/5/2020, 12:32:44 PM] [EufySecurityHomebridgePlugin] Config {
  username: 'me@example.com',
  password: 'PASSWORDHERE',
  enablePush: false,
  platform: 'EufySecurityHomebridgePlugin'
}
[12/5/2020, 12:32:44 PM] [EufySecurityHomebridgePlugin] Finished initializing platform: undefined
[12/5/2020, 12:32:44 PM] [EufySecurityHomebridgePlugin] push client disabled
[12/5/2020, 12:32:44 PM] [EufySecurityHomebridgePlugin] Executed didFinishLaunching callback

(node:65778) UnhandledPromiseRejectionWarning: HTTPError: Response code 401 (Unauthorized)
    at Request.<anonymous> (/Users/dshafik/.nvm/versions/node/v14.15.1/lib/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/as-promise/index.js:117:42)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:65778) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:65778) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

One issue that can be easily fixed is:

> [12/5/2020, 12:32:44 PM] [EufySecurityHomebridgePlugin] Finished initializing platform: undefined


```diff
--       this.log.debug("Finished initializing platform:", this.config.name);
++       this.log.debug("Finished initializing platform:", this.config.platform);
```

But that's not going to impact anything of consequence.

**Comment by DaveM8686** [comment]
Still nothing for me, with push on or off.

**Comment by DaveM8686** [comment]
Is there a dependent package for this? Like a camera or doorbell one?

**Comment by LikeYourMommaToldYou** [comment]
I have the newest version. 0.1.2 and my logs look like this.
EufySecurityHomebridgePlugin] failed to confirm push token
(node:93803) UnhandledPromiseRejectionWarning: Error: Request failed: https://security-app.eufylife.com/v1/passport/login -> 26006 - 504 Gateway Timeout
    at /usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:24:15
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:5:58)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:93803) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:93803) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by cmfrazier** [comment]
@raphb83 how are you running Homebridge? Pi or other?

**Comment by LikeYourMommaToldYou** [comment]
I have it running on an iMac.

**Comment by awhelme** [comment]
I have the same error running on a Mac Mini

**Comment by dshafik** [comment]
I am also using Mac, specifically a 2018 MbP with macOS Big Sur

**Comment by Techlover22** [comment]
I have it running on my 2020 13" MacBook Pro running Big Sur.

**Comment by raphb83** [comment]
> @raphb83 how are you running Homebridge? Pi or other? 

@cmfrazier running on hoobs. Only issue right now that there is no refresh every 10 seconds

**Comment by awhelme** [comment]
Now I'm getting:

failed to register push token HTTPError: Response code 401 (Unauthorized)
    at Request.<anonymous> (/usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/as-promise/index.js:117:42)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  code: undefined,
  timings: {
    start: 1607263228568,
    socket: 1607263228568,
    lookup: 1607263228569,
    connect: 1607263228647,
    secureConnect: 1607263228708,
    upload: 1607263228708,
    response: 1607263228768,
    end: 1607263228768,
    error: undefined,
    abort: undefined,
    phases: {
      wait: 0,
      dns: 1,
      tcp: 78,
      tls: 61,
      request: 0,
      firstByte: 60,
      download: 0,
      total: 200
    }
  }
}

Sounds like a password issue maybe? I've tried multiple accounts. Thanks for the help!

**Comment by mrota83** [comment]
The plugin don't work with homebridge in docker synology. I have the eufy wired doorbell always in error.

**Comment by birkir** [comment]
> The plugin don't work with homebridge in docker synology :(. I have the eufy wired doorbell always in error and do not show in ios home app.

I have this plugin running on Synology.

Battery doorbell (with HomeBase 2). No support for wired yet.

**Comment by LikeYourMommaToldYou** [comment]
I have the battery doorbell, via Homebase 2 as well. In my case Homebridge is running (V1.2.3) on MacOS Big Sur on an iMac. 

The Homebase is in HomeKit and also has an outside battery powered camera that has native HomeKit support. 

As you can see above I am getting an error initializing the connection. I have tried with Push turned on and off.

**Comment by mrota83** [comment]
> > The plugin don't work with homebridge in docker synology :(. I have the eufy wired doorbell always in error and do not show in ios home app.
> 
> I have this plugin running on Synology.
> 
> Battery doorbell (with HomeBase 2). No support for wired yet.

I thought the wired version was supported. Do you have plan to support wired version in future ? 
Thank you for your work !

**Comment by cmfrazier** [comment]
@raphb83 appears to have his wired version working based on his comment. I guess we should clarify, is it the battery version but it's wired or is it the original wired doorbell?

**Comment by raphb83** [comment]
> @raphb83 appears to have his wired version working based on his comment. I guess we should clarify, is it the battery version but it's wired or is it the original wired doorbell?

It's the original wired doorbell ([](https://www.eufylife.com/de/products/variant/2k-videoturklingel-kabelgebunden/T8200311))

**Comment by cmfrazier** [comment]
@raphb83 would you mind posting your debug logs and config since your's appears to be working.

**Comment by royaldunlin** [comment]
Just installed this plugin today and am getting the following error:

```
(node:2439) UnhandledPromiseRejectionWarning: RequestError: getaddrinfo ENOTFOUND security-app.eufylife.com
    at ClientRequest.<anonymous> (/usr/lib/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/core/index.js:953:111)
    at Object.onceWrapper (events.js:421:26)
    at ClientRequest.emit (events.js:326:22)
    at ClientRequest.origin.emit (/usr/lib/node_modules/homebridge-plugin-eufy-security/node_modules/@szmarczak/http-timer/dist/source/index.js:39:20)
    at TLSSocket.socketErrorListener (_http_client.js:427:9)
    at TLSSocket.emit (events.js:314:20)
    at emitErrorNT (internal/streams/destroy.js:92:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:66:26)
(node:2439) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:2439) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I'm running on an Ubuntu VM.

**Comment by sccrrich** [comment]
I too am getting an error. Mine states “failed to confirm push token”. No eufy doorbell or cameras show up. Any ideas?

**Comment by cmfrazier** [comment]
No change in logs using 0.1.3

**Comment by royaldunlin** [comment]
I now have a new error message with the update to 0.1.3:

```
(node:23082) UnhandledPromiseRejectionWarning: HTTPError: Response code 401 (Unauthorized)
    at Request.<anonymous> (/usr/lib/node_modules/homebridge-plugin-eufy-security/node_modules/got/dist/source/as-promise/index.js:117:42)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
(node:23082) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:23082) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by raphb83** [comment]
> @raphb83 would you mind posting your debug logs and config since your's appears to be working.
> 
> 

I will post debug logs and config as soon a I manage to run my hoobs in debug mode😬

**Comment by cmfrazier** [comment]
> > @raphb83 would you mind posting your debug logs and config since your's appears to be working.
> 
> I will post debug logs and config as soon a I manage to run my hoobs in debug mode😬

HOOBS runs as a service. You need to shutdown the service before starting HOOBS in debug mode.
`sudo hoobs service stop`
Then `hoobs -d`
After you get the information you need `sudo hoobs service start`
[HOOBS CLI CMD List](https://support.hoobs.org/docs/5ea6858d0ab68b0344e872e3)
[Source](https://www.reddit.com/r/HOOBS/comments/fhqou4/debug_help/)

**Comment by raphb83** [comment]
> > > @raphb83 would you mind posting your debug logs and config since your's appears to be working.
> > 
> > 
> > I will post debug logs and config as soon a I manage to run my hoobs in debug mode😬
> 
> @raphb83 HOOBS runs as a service. You need to shutdown the service before starting HOOBS in debug mode.
> `sudo hoobs service stop`
> Then `hoobs -d`
> After you get the information you need `sudo hoobs service start`
> [HOOBS CLI CMD List](https://support.hoobs.org/docs/5ea6858d0ab68b0344e872e3)
> [Source](https://www.reddit.com/r/HOOBS/comments/fhqou4/debug_help/)

I am quite not sure, if this is really the log you asked for - but here it is:

```
Last login: Wed Dec  9 17:15:11 2020 from 192.168.1.XXX
-bash: /etc/profile: line 46: unexpected EOF while looking for matching `"'
-bash: /etc/profile: line 65: syntax error: unexpected end of file
hoobs@hoobs:~ $ hoobs -D
HOOBS listening on port 8080.
Loaded plugin 'homebridge-plugin-eufy-security'
[12/9/2020, 17:17:44] Registering platform 'homebridge-plugin-eufy-security.EufySecurityHomebridgePlugin'
Loading 4 platforms...
[Eufy] Initializing EufySecurityHomebridgePlugin platform...
[Eufy] Loading accessory from cache: DoorBell
[Eufy] push client disabled
Bridge is running on port 51826.
[Eufy] Restoring existing accessory from cache: Doorbell
```

And here's my config:

{
    "server": {
        "polling_seconds": 1,
        "home_setup_id": "...",
        "port": 8080,
        "origin": "*",
        "autostart": 10
    },
    "client": {
        "country_code": "DE",
        "default_route": "status",
        "locale": "de",
        "inactive_logoff": 30,
        "postal_code": "...",
        "theme": "hoobs-dark",
        "temp_units": "celsius"
    },
    "bridge": {
        "pin": ".....",
        "port": .....,
        "username": ".....",
        "name": "HOOBS"
    },
    "description": "",
    "ports": {},
    "accessories": [],
    "platforms": [
        {
            "platform": "EufySecurityHomebridgePlugin",
            "name": "Eufy",
            "username": "my-email",
            "password": "my-password",
            "plugin_map": {
                "plugin_name": "homebridge-plugin-eufy-security"
            }
        }
    ]
}

**Comment by cmfrazier** [comment]
@raphb83 Interesting. Is this the only plugin you have?

**Comment by raphb83** [comment]
> @raphb83 Interesting. Is this the only plugin you have?

No. I got the following plugins too:
- Camera Ffmpeg
- Homematic
- Secvest

Probabely it is interresting that I have got some eufy cam 2 pro connected via Eufy HomeBase2 with HomeKit?

**Comment by cmfrazier** [comment]
@raphb83 is the doorbell connected to the homebase2 as well?

**Comment by raphb83** [comment]
> @raphb83 is the doorbell connected to the homebase2 as well?

No. The doorbell is shown in the eufy app, but not connected with the homebase2 due to not supported

**Comment by cmfrazier** [comment]
@raphb83 interesting yours is working and mine is not. 
what version of NodeJS are you running? `node -v`
I'm on 12.16.3.
Also what version of homebridge?

**Comment by jdraxgit** [comment]
I'm having similar issues as others here, my homebridge comes pre-installed with node is 14.15.1. Using homebridge 1.2.4 on an Intel NUC. Have also tried on a rpi 3b+ on homebridge 1.1.6 (also node is 14.15.1). Is there a way to downgrade the node js?

**Comment by birkir** [comment]
Should work on node 10 through 15
https://github.com/birkir/homebridge-plugin-eufy-security/actions/runs/409417668

**Comment by birkir** [comment]
I think i figured out the issue. I was discovering devices through hubs, I didn't know devices worked without one.

So now I discover devices without going through hubs.

Try 0.1.4

**Comment by cmfrazier** [comment]
@birkir I updated to the latest LTS node with no changes. 
```
[10/12/2020, 08:51:35] [EufySecurityHomebridgePlugin] Executed didFinishLaunching callback
(node:18983) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:18983) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 shutdown listeners added to [HomebridgeAPI]. Use emitter.setMaxListeners() to increase limit
(node:18983) UnhandledPromiseRejectionWarning: Error: Request failed: https://security-app.eufylife.com/v1/passport/login -> 26006 - 504 Gateway Timeout
    at /usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:24:15
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:5:58)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(node:18983) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
(node:18983) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
When I try to load `https://security-app.eufylife.com/v1/passport/login` in Safari I get the following:
<img width="1241" alt="Screen Shot 2020-12-10 at 10 51 44 AM" src="https://user-images.githubusercontent.com/20451193/101803047-b855ae00-3ad5-11eb-8cdb-22ebfffa6cd3.png">

**Comment by cmfrazier** [comment]
It's all above my head but it looks like lots of good active API work going on here:
[FuzzyMistborn/python-eufy-security](https://github.com/FuzzyMistborn/python-eufy-security)
[JanLoebel/eufy-node-client](https://github.com/JanLoebel/eufy-node-client)

**Comment by jdraxgit** [comment]
> I think i figured out the issue. I was discovering devices through hubs, I didn't know devices worked without one.
> 
> 
> 
> So now I discover devices without going through hubs.
> 
> 
> 
> Try 0.1.4

Tried 0.1.4 still nothing appearing... 

Logs: 

[EufySecurityHomebridgePlugin] failed to register push token Error: Request failed: https://security-app.eufylife.com/v1/passport/login -> 100027 - Failed to request.
    at /homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:24:15
    at Generator.next (<anonymous>)
    at fulfilled (/homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:5:58)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)

**Comment by LikeYourMommaToldYou** [comment]
I have updated to the latest with no luck. My T8210 Doorbell requires the connection to Homebase 2 it will not work standalone. 

Is there a version of the doorbell that does not require Homebase connection? A wired doorbell perhaps? Mine is wireless.

**Comment by jdraxgit** [comment]
> I have updated to the latest with no luck. My T8210 Doorbell requires the connection to Homebase 2 it will not work standalone. 
> 
> Is there a version of the doorbell that does not require Homebase connection? A wired doorbell perhaps? Mine is wireless. 

This is the same mode I have and have never gotten it to work with any version of this plugin. Not giving up though!

**Comment by cmfrazier** [comment]
> I think i figured out the issue. I was discovering devices through hubs, I didn't know devices worked without one.
> 
> So now I discover devices without going through hubs.
> 
> Try 0.1.4

No luck.
```
[10/12/2020, 12:26:42] [EufySecurityHomebridgePlugin] failed to register push token Error: Request failed: https://security-app.eufylife.com/v1/passport/login -> 26006 - 504 Gateway Timeout
    at /usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:24:15
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:5:58)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
[10/12/2020, 12:26:42] [EufySecurityHomebridgePlugin] registered at eufy with: ***redactedtokenofsomesort***
```

**Comment by raphb83** [comment]
> I have updated to the latest with no luck. My T8210 Doorbell requires the connection to Homebase 2 it will not work standalone. 
> 
> Is there a version of the doorbell that does not require Homebase connection? A wired doorbell perhaps? Mine is wireless. 

Mine - Version T8200 (this is the wired version) - is working well - without HomeBase2. T8200 does not require connection to HomeBase2
![image](https://user-images.githubusercontent.com/71670682/101820573-18c10b80-3b27-11eb-8f2b-33e8a2fded25.jpeg)

**Comment by raphb83** [comment]
> @raphb83 interesting yours is working and mine is not. 
> 
> what version of NodeJS are you running? `node -v`
> 
> I'm on 12.16.3.
> 
> Also what version of homebridge?
> 
> 

I'm running on
- HOOBS 3 - Version	3.3.1
- Node Version	12.16.3

**Comment by LikeYourMommaToldYou** [comment]
Interesting, I think perhaps this plugin only works on the wired doorbell. 

My info:
![image](https://user-images.githubusercontent.com/57150004/101821338-d6330b00-3af5-11eb-9467-93479099cc0e.jpeg)

**Comment by awhelme** [comment]
The error message changed for me on Wired Doorbell T8201 that does not hate the homebase2:

(node:3341) UnhandledPromiseRejectionWarning: Error: Request failed: https://mysecurity.eufylife.com/api/v1/passport/login -> 100027 - Failed to request.
    at /usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:24:15
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-plugin-eufy-security/node_modules/eufy-node-client/build/http/http.utils.js:5:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
(node:3341) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:3341) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by karesake** [comment]
I uninstalled the plugin for testing and reinstalled it. Now the camera doesn't appear in the Home app at all for me either. HomeBridge also no longer shows anything in the devices.

**Comment by djrams10** [comment]
I have the same problem. do not log out of your Eufy account from the application because for me it is impossible for me to reconnect to the ios Eufy application. Looks like there has been too much demand. I had to contact Eufy support my account is blocked.

![IMG_213D1B018C06-1 3](https://user-images.githubusercontent.com/28505910/101977894-38545300-3c51-11eb-8885-db2093a1ed04.jpeg)

**Comment by djrams10** [comment]
#10

**Comment by djrams10** [comment]
#10 registering a push token timed out

**Comment by jmwarren85** [comment]
My eufy account was also blocked by this plugin failing

**Comment by DaveM8686** [comment]
@birkir you mentioned you have the battery doorbell running on Synology?  To clarify, which HomeBridge are you using?

I'm using:
- T8210 2K Battery Doorbell with Home Base 2
- Synology NAS running homebridge-config-ui-x via Docker
- A separate Eufy account created exclusively for this with no 2FA and no admin rights, view only.

I'm getting nothing.

**Comment by birkir** [comment]
> @birkir you mentioned you have the battery doorbell running on Synology?  To clarify, which HomeBridge are you using?
> 
> 
> 
> I'm using:
> 
> - T8210 2K Battery Doorbell with Home Base 2
> 
> - Synology NAS running homebridge-config-ui-x via Docker
> 
> - A separate Eufy account created exclusively for this with no 2FA and no admin rights, view only.
> 
> 
> 
> I'm getting nothing.  

Same as you, but no secondary account.
Is the account verified and such?

**Comment by DaveM8686** [comment]
> > @birkir you mentioned you have the battery doorbell running on Synology?  To clarify, which HomeBridge are you using?
> > I'm using:
> > 
> > * T8210 2K Battery Doorbell with Home Base 2
> > * Synology NAS running homebridge-config-ui-x via Docker
> > * A separate Eufy account created exclusively for this with no 2FA and no admin rights, view only.
> > 
> > I'm getting nothing.
> 
> Same as you, but no secondary account.
> Is the account verified and such?

Yep, definitely.  I even tried upgrading it to admin to see if that helped, and nothing.  I then dropped it back down to guest to see if it had made a difference, and again nothing.

I also have the motion detection turned off on the doorbell, but the camera on.  I've tried turning on the motion detection and that didn't help either.

What settings do you have on your doorbell, if you don't mind me asking? IE, video quality, firmware version, power management settings, etc.  I'm willing to try pretty much anything at this point.

**Comment by DaveM8686** [comment]
> > @birkir you mentioned you have the battery doorbell running on Synology?  To clarify, which HomeBridge are you using?
> > 
> > 
> > 
> > I'm using:
> > 
> > - T8210 2K Battery Doorbell with Home Base 2
> > 
> > - Synology NAS running homebridge-config-ui-x via Docker
> > 
> > - A separate Eufy account created exclusively for this with no 2FA and no admin rights, view only.
> > 
> > 
> > 
> > I'm getting nothing.  
> 
> Same as you, but no secondary account.
> Is the account verified and such?

I forgot to ask, what other plugins do you have on HomeBridge? I'm wondering if there's a dependency somewhere that we haven't realised.

**Comment by djrams10** [comment]
> > > @birkir you mentioned you have the battery doorbell running on Synology?  To clarify, which HomeBridge are you using?
> > > I'm using:
> > > 
> > > * T8210 2K Battery Doorbell with Home Base 2
> > > * Synology NAS running homebridge-config-ui-x via Docker
> > > * A separate Eufy account created exclusively for this with no 2FA and no admin rights, view only.
> > > 
> > > I'm getting nothing.
> > 
> > 
> > Same as you, but no secondary account.
> > Is the account verified and such?
> 
> I forgot to ask, what other plugins do you have on HomeBridge? I'm wondering if there's a dependency somewhere that we haven't realised.

Nothing else.

**Comment by mrota83** [comment]
Maybe it doesn't work with the wired version t8200 because we need to set the resolution to 1080p in the eufy app before install the plugin in homebridge ?

**Comment by DaveM8686** [comment]
(node:7580) UnhandledPromiseRejectionWarning: Error: Request failed: https://mysecurity.eufylife.com/api/v1/passport/login -> 100027 - Failed to request.

I'm getting this.  That website link shows a 404, but I'm not sure if it's supposed to do that because it's an api, or if that's what's causing the problem.

**Comment by neumanns** [comment]
> (node:7580) UnhandledPromiseRejectionWarning: Error: Request failed: https://mysecurity.eufylife.com/api/v1/passport/login -> 100027 - Failed to request.
> 
> I'm getting this. That website link shows a 404, but I'm not sure if it's supposed to do that because it's an api, or if that's what's causing the problem.

the problem is, that's your account ist blocked ! for the next 48 hours.
you cann test it! go to:

https://mysecurity.eufylife.com
logged out and than test if can can login

**Comment by DaveM8686** [comment]
> > (node:7580) UnhandledPromiseRejectionWarning: Error: Request failed: https://mysecurity.eufylife.com/api/v1/passport/login -> 100027 - Failed to request.
> > I'm getting this. That website link shows a 404, but I'm not sure if it's supposed to do that because it's an api, or if that's what's causing the problem.
> 
> the problem is, that's your account ist blocked ! for the next 48 hours.
> you cann test it! go to:
> 
> https://mysecurity.eufylife.com
> logged out and than test if can can login

Nope, I can log in and out no problem.

**Comment by DaveM8686** [comment]
Something that was just pointed out to me: the Eufy App will only allow four cameras per Home Base 2 to move to HomeKit.  I've already got 4 in there.  Surely that couldn't be the problem for Homebridge as well, right?

For everyone who has it working and who doesn't, how many cameras do you have on that same Home Base that are already in HomeKit?

**Comment by djrams10** [comment]
> Quelque chose qui vient de me faire remarquer : l'application Eufy ne permettra qu'à quatre caméras par Home Base 2 de passer à HomeKit. J'en ai déjà 4 là-dedans. Cela ne pourrait sûrement pas être le problème pour Homebridge aussi, n'est-ce pas ?
> 
> Pour tous ceux qui l'ont fait fonctionner et qui ne fonctionnent pas, combien de caméras avez-vous sur cette même base d'accueil qui sont déjà dans HomeKit ?

I only have one and it doesn’t work in homebridge

**Comment by LikeYourMommaToldYou** [comment]
I have 2 cameras including the Wireless doorbell on my Homebase 2. When I got the plugin working it brought both cameras into HomeKit, actually duplicating one of them that was already supported directly by HomeKit. 

My issue with getting the plugin working turned out to be an incorrect install of ffmpeg. 

On MacOS the best way to install ffmpeg is via Homebrew. 

Brew install ffmpeg

This corrected the issue I was having and I reinstalled the plugin and now it works.

**Comment by DaveM8686** [comment]
Hey, 0.1.6 did the trick!  Awesome!  Except, like @LikeYourMommaToldYou, it's brought in duplicates of all my Eufy cameras that were already in there.  Hopefully I can remove those as cached devices.

**Comment by DaveM8686** [comment]
Ok, it let me remove them, but now the doorbell is not responding.  One thing at a time, I guess...

**Comment by mrota83** [comment]
Ok so after create another invite account no 2fa activated and updated to latest version 0.16 i have a the t8200 wired doorbell no homebase and its finally working !!!

**Comment by gchattha** [comment]
> Ok so after create another invite account no 2fa activated and updated to latest version 0.16 i have a the t8200 wired doorbell no homebase and its finally working !!!
> 
> Could be fantastic if we can have the stream in 1080p in home app instead 720p :)

I have same setup but my Apple Home app keeps showing "No Response" for the doorbell though and Homebridge shows "Unhandled Promise Rejection" and 401 error

**Comment by mrota83** [comment]
> > Ok so after create another invite account no 2fa activated and updated to latest version 0.16 i have a the t8200 wired doorbell no homebase and its finally working !!!
> > Could be fantastic if we can have the stream in 1080p in home app instead 720p :)
> 
> I have same setup but my Apple Home app keeps showing "No Response" for the doorbell though and Homebridge shows "Unhandled Promise Rejection" and 401 error

Do you have a good wifi ??  sometimes when i get no response in home app its because the wifi is not optimal.

**Comment by gchattha** [comment]
> > > Ok so after create another invite account no 2fa activated and updated to latest version 0.16 i have a the t8200 wired doorbell no homebase and its finally working !!!
> > > Could be fantastic if we can have the stream in 1080p in home app instead 720p :)
> > 
> > 
> > I have same setup but my Apple Home app keeps showing "No Response" for the doorbell though and Homebridge shows "Unhandled Promise Rejection" and 401 error
> 
> Do you have a good wifi ?? sometimes when i get no response in home app its because the wifi is not optimal.

Yup wifi is great, have 2 other Eufy Indoor Cams using the FFMpeg plug in further away and they are working fine. Uninstalled and reinstalled the Eufy Security plugin and now getting a different error in Homebridge, Home app still shows no response:

[12/23/2020, 4:40:43 PM] [EufySecurityHomebridgePlugin] An error occurred while making snapshot request: Error: spawn /homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/ffmpeg-for-homebridge/ffmpeg EACCES Doorbell

**Comment by mrota83** [comment]
> > > > Ok so after create another invite account no 2fa activated and updated to latest version 0.16 i have a the t8200 wired doorbell no homebase and its finally working !!!
> > > > Could be fantastic if we can have the stream in 1080p in home app instead 720p :)
> > > 
> > > 
> > > I have same setup but my Apple Home app keeps showing "No Response" for the doorbell though and Homebridge shows "Unhandled Promise Rejection" and 401 error
> > 
> > 
> > Do you have a good wifi ?? sometimes when i get no response in home app its because the wifi is not optimal.
> 
> Yup wifi is great, have 2 other Eufy Indoor Cams using the FFMpeg plug in further away and they are working fine. Uninstalled and reinstalled the Eufy Security plugin and now getting a different error in Homebridge, Home app still shows no response:
> 
> [12/23/2020, 4:40:43 PM] [EufySecurityHomebridgePlugin] An error occurred while making snapshot request: Error: spawn /homebridge/node_modules/homebridge-plugin-eufy-security/node_modules/ffmpeg-for-homebridge/ffmpeg EACCES Doorbell

had you already installed the ffmpeg plugin Before installing the eufy plugin ?
I didn’t have installed ffmpeg. When i have installed the eufy plug-in it seems to come bundled with.
